mongodb/src/main/java/org/hibernate/ogm/datastore/mongodb/query/parsing/nativequery/impl/MongoDBQueryDescriptorBuilder.java

```diff
@@ -16,13 +16,18 @@
  * Builder for {@link MongoDBQueryDescriptor}s.
  *
  * @author Gunnar Morling
+ * @author Thorsten Möller
  */
 public class MongoDBQueryDescriptorBuilder {
 
 	private String collection;
 	private Operation operation;
-	private String criteria;
-	private String projection;
+	private String criteria;   // Overloaded to be the 'document' for a FINDANDMODIFY query (which is a kind of criteria),
+	                           //                      document or array of documents to insert for an INSERT query.
+	private String projection; // Overloaded to be the 'update' for Operation.UPDATE.
```

Comment:
I don't agree. Consider a system that creates gazillions of queries per hour. Such systems exist. This will create (and garbage collect) gazillions of MongoDBQueryDescriptorBuilder instances. Being memory-efficient starts with such (supposedly unimportant) things. You decide.